app/src/pages/playground/Playground.tsx

```diff
@@ -50,35 +60,132 @@ function AddPromptButton() {
+    flex: 1 1 auto;
+    .ac-accordion-item {
+      height: 100%;
+      overflow: hidden;
+      flex: 1 1 auto;
+      .ac-accordion-itemContent {
+        height: 100%;
+        overflow: hidden;
+        flex: 1 1 auto;
+        & > * {
+          height: 100%;
+          flex: 1 1 auto;
+          overflow: auto;
+          box-sizing: border-box;
+          // Fix padding issue with flexbox
+          padding-bottom: 57px !important;
+        }
+      }
+    }
+  }
+`;
+
+const playgroundInputOutputPanelContentCSS = css`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  overflow: auto;
+`;
+
+function PlaygroundContent() {
   const instances = usePlaygroundContext((state) => state.instances);
+  const numInstances = instances.length;
+  const isSingleInstance = numInstances === 1;
+
   return (
-    <Flex direction="row" alignItems="stretch" height="100%" flex="1 1 auto">
-      <PanelGroup direction="horizontal">
-        {instances.map((instance, i) => (
-          <Fragment key={i}>
-            {i !== 0 && <PanelResizeHandle css={resizeHandleCSS} />}
-            <Panel defaultSize={50}>
-              <PlaygroundInstance key={i} playgroundInstanceId={instance.id} />
-            </Panel>
-          </Fragment>
-        ))}
-      </PanelGroup>
-    </Flex>
+    <PanelGroup
+      direction={isSingleInstance ? "horizontal" : "vertical"}
+      autoSaveId={
+        isSingleInstance ? "playground-horizontal" : "playground-vertical"
+      }
+    >
+      <Panel>
+        <div css={playgroundPromptPanelContentCSS}>
+          <Accordion>
+            <AccordionItem
+              title="Prompts"
+              id="prompts"
+              extra={<AddPromptButton />}
+            >
+              <View height="100%" padding="size-200">
+                <Flex direction="row" gap="size-100">
+                  {instances.map((instance, i) => (
+                    <div
+                      key={i}
+                      css={css`
+                        flex: 1 1 0px;
+                      `}
+                    >
+                      <PlaygroundTemplate
+                        key={i}
+                        playgroundInstanceId={instance.id}
+                      />
+                    </div>
+                  ))}
+                </Flex>
+              </View>
+            </AccordionItem>
+          </Accordion>
+        </div>
+      </Panel>
+      <PanelResizeHandle css={resizeHandleCSS} />
+      <Panel>
+        <div css={playgroundInputOutputPanelContentCSS}>
+          <Accordion>
+            <AccordionItem title="Inputs" id="input">
+              <View padding="size-200">Inputs go here</View>
+            </AccordionItem>
+            <AccordionItem title="Output" id="output">
+              <View padding="size-200" height="100%">
+                <Flex direction="row" gap="size-200">
+                  {instances.map((instance, i) => (
+                    <View key={i} flex="1 1 0px">
```

Comment:
key of instanceId here to avoid index as key which might cause issues when adding and deleting instances, same with above

Reply:
good catch